src/dns.mli

```diff
@@ -791,6 +791,35 @@ module Edns : sig
       buffer. *)
 end
 
+(** Loc
+
+    A locator record (LOC) is used to express location information associated with domain. *)
+module Loc : sig
+  type t = {
+    lat : int32;
+    long : int32;
```

Comment:
can lat/long simply be `int` to avoid boxing the value? OCaml guarantees at least 31 bits for an `int`

Reply:
Using `int` doesn't seem to deal with negatives properly, I think because 32nd bit is used to encode the direction of the lat/long.

Using an `int` version of `lat_long_parse` and `lat_long_print`:
```
let lat_long_parse_int_version ((deg, min, sec), dir) =
    (Int.shift_left 1 31) + (
      (((Int32.to_int deg * 60) + Int32.to_int min) * 60) * 1000 +
      Int.of_float (sec *. 1000.)
    ) * if dir then 1 else -1

let lat_long_print_int_version lat_long =
    let dir = lat_long < 0 in
    let lat_long = Int.abs lat_long in
    let lat_long = (Int.shift_left 1 31) - lat_long in
    let sec = Float.of_int (lat_long mod (60 * 1000)) /. 1000. in
    let min = (lat_long / (1000 * 60)) mod 60 in
    let deg = (lat_long / (1000 * 60 * 60)) mod 60 in
    deg, min, sec, dir
```

With 0 0 1 N latitude we get:
```
# lat_long_parse_int_version ((0l, 0l, 1.), true);;
- : int = 2147484648
```

And as the directionality is encoded by the sign (e.g. the direction is encoded in the first bit) this messes up our decoding:
```
# lat_long_print_int_version 2147484648;;
- : int * int * float * bool = (0, 0, -1., false)
```

We should never have a negative deg/min/sec (as specified in [RFC1876 section 3](https://datatracker.ietf.org/doc/html/rfc1876#section-3)).

In contrast, Int32 deals with the negative properly:
```
# lat_long_parse ((0l, 0l, 1.), true);;
- : int32 = -2147482648l

# lat_long_print (-2147482648l);;
- : int * int * float * bool = (0, 0, 1., true)
```

Note that converting the `int` to an `int32` gives the desired negative value, which is what we're doing at the end of `lat_long_parse_int_version`:
```
Int32.of_int 2147484648;;
- : int32 = -2147482648l
```

Reply:
maybe it would be better to explicitly use the `Int32` arithmetic operations?

Reply:
Looking at this with a fresh eye, there was a bug in my `lat_long_print` decoding that replied on the leftmost bit being negative, but actually this should only be the case after 2^32 is subtracted from the value.

TL;DR: using `int` is possible and desirable!